### PR TITLE
Add more robust unicode handling

### PR DESCRIPTION
**Pull request by lsetiawan**
## Overview

This PR addresses https://github.com/ODM2/WOFpy/issues/148#issuecomment-326420842 issue regarding encoding. It seems that I missed some utf-8 encodings in `WaterML_1_1.py` and `quote_xml` is now made to use unicode codec to encode the characters coming in, which then will be encoded again to `utf-8` so that it can be shown to the screen.

### CODE REVIEW

**Comment by emiliom** [comment]
@ocefpaf, I added you as the reviewer instead of me, since you're the unicode pro here, as demonstrated by [your comment](148#issuecomment-326565328) earlier today :wink:

Thanks, guys!

**Comment by lsetiawan** [comment]
Gah! @ocefpaf please help. What am i doing wrong.. it works in my computer... :(

**Comment by emiliom** [comment]
I'm glad I passed the buck to @ocefpaf -- even before the tests failed :innocent:

**Comment by ocefpaf** [comment]
I'll take a look in a few hours. (Trying to wrap the `choropleth` stuff in `folium` so we can ship it.)

**Comment by emiliom** [comment]
> I'll take a look in a few hours. 

No rush. The release is now out :) It can wait till early next week or so, if that helps you with getting Folium out.

> (Trying to wrap the choropleth stuff in folium so we can ship it.)

Good luck!! Looking forward to it.

File: wof/WaterML_1_1.py [hunk 1]
```diff
@@ -402,7 +402,7 @@ def quote_xml(inStr):
         pos = mo.end()
     s3 = s1[pos:]
     s2 += quote_xml_aux(s3)
-    return s2.decode(ExternalEncoding)
+    return s2.encode(ExternalEncoding).decode(ExternalEncoding)
```
Comment (ocefpaf) [inline wof/WaterML_1_1.py]: @lsetiawan the `ExternalEncoding` is `utf-8`, correct?

We are playing the dangerous game of encode/decode whack-a-mole :smile: 

The `WaterML_1_1.py` module is gigantic, >7500 lines :scream:  Any operation here is non-trivial and we are barely capable of doing damage control.

The usual logic that we should try is:

- read everything as bytes (xml, etc), if it is not already bytes we should encode it;
- use bytes internally in the API;
- decode only to write.

I must confess that I have no idea what we are reading and writing here. Maybe we need a live call to discuss a long term goal for this, specially if Python 3 is you ODM2 sights.

Comment (lsetiawan) [reply wof/WaterML_1_1.py]: > @lsetiawan the ExternalEncoding is utf-8, correct?

Yes it is.

Comment (lsetiawan) [reply wof/WaterML_1_1.py]: > I must confess that I have no idea what we are reading and writing here.

We are reading values from a database and writing Water ML XML representation. The failures comes when trying to writing that into XML.

Comment (ocefpaf) [reply wof/WaterML_1_1.py]: I am trying something here... Let's see... Hold on!

Comment (emiliom) [reply wof/WaterML_1_1.py]: @ocefpaf: seriously, there's no rush on this!! Go back to folium to finish it off and ship the release! Unless you enjoy distractions :stuck_out_tongue_winking_eye: 

We can reconnect next week, to discuss both these unicode issues and longer term goals.

**Comment by ocefpaf** [comment]
> @ocefpaf: seriously, there's no rush on this!! Go back to folium to finish it off and ship the release! Unless you enjoy distractions :stuck_out_tongue_winking_eye: 

Too late. I won't be able to sleep if I don't fix this :smile: (or at least find a strategy for a long term plan.)

**Comment by emiliom** [comment]
> Too late. I won't be able to sleep if I don't fix this :smile: (or at least find a strategy for a long term plan.)

Ok. I can't help you, then. But maybe I can look up online a good therapist to recommend, in your town ..

**Comment by lsetiawan** [comment]
@ocefpaf is handling this in #188. Closing...

**Pull request closed without merging**